src/util/emitUtils.js

```diff
@@ -578,22 +580,29 @@ function emitToClients(io, nsp, rooms, key, obj) {
  */
 function doEmit(nsp, key, obj) {
   const newObjectAsString = getNewObjAsString(key, obj); // { key: {new: obj }}
-  const numClients = Object.values(nsp.connected).length;
-  if (key.startsWith('refocus.internal.realtime.sample')) {
-    tracker.trackEmit(obj.name, obj.updatedAt, numClients);
-  }
   pubSubStats.trackEmit(key, obj);
-  if (!toggle.isFeatureEnabled('enableClientStats')) { // dont enter the else if its not a sample
-    nsp.emit(key, newObjectAsString);
-  } else if (key.startsWith('refocus.internal.realtime.sample')) {
+  if (toggle.isFeatureEnabled('enableClientStats') && key.startsWith('refocus.internal.realtime.sample')) { // dont enter the else if its not a sample
+    let numClients = 0;
+    const rooms = nsp.rooms.slice(0);
+    nsp.rooms = [];
     Object.values(nsp.connected)
+      .filter((socket) => rooms.includes(Object.keys(socket.rooms)[1]))
       .forEach((socket) => {
+        numClients++;
```

Comment:
Seems like it would be cleaner to save the filter result as a variable and do trackEmit(sockets.length), to avoid having to track and increment a separate numClients variable.

Reply:
Added inside `doEmitWithTracking`